### PR TITLE
feat: add parse maven repository and save to sqlite db

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
add parse maven repository and save to sqlite db

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package metadata
```
Comment (knqyf263) [inline metadata/metadata.go:1]: Why is this package located outside of pkg/?

File: pkg/crawler/crawler.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+loop:
+	for {
+		select {
+		case url, ok := <-c.urlCh:
+			count++
+			if count%1000 == 0 {
+				log.Printf("Count: %d", count)
+			}
+			if !ok { // channel is closed
+				// save metadata
```
Comment (knqyf263) [inline pkg/crawler/crawler.go:101]: We should put it outside of the for-loop.

File: pkg/crawler/crawler.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+				metaDB := metadata.Metadata{
+					Version:    db.SchemaVersion,
+					NextUpdate: c.clock.Now().UTC().Add(db.UpdateInterval),
+					UpdatedAt:  c.clock.Now().UTC(),
+				}
+				err := metadata.Update(metaDB)
+				if err != nil {
+					close(c.urlCh)
```
Comment (knqyf263) [inline pkg/crawler/crawler.go:109]: urlCh is already closed here

File: pkg/crawler/crawler.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+	c.wg.Add(1)
+
+	go func() {
+		c.wg.Wait()
+		close(c.indexCh)
+		close(c.urlCh)
+		c.saveIndexesWG.Wait()
+	}()
+
+	go func() { // function for saving indexes in the database
```
Comment (knqyf263) [inline pkg/crawler/crawler.go:69]: Do we need to run it in goroutine? I think DB update is fast enough compared to HTTP.

Comment (DmitriyLewen) [reply pkg/crawler/crawler.go:69]: we need 2 parallel loops:
1) to parse all URLS(https://github.com/aquasecurity/trivy-java-db/pull/1/files#diff-526a7142f7b15a72eab215dcfa540f5b6d502951a2f9f3b34dd900013e82c675R92)
2) to save indexes to DB.

1 loop works in main gorutine
Therefore, I run saved indexes in separate goroutine.

File: pkg/crawler/crawler.go (hunk 3 above)
Comment (DmitriyLewen) [reply pkg/crawler/crawler.go:109]: nice catch! fixed

File: metadata/metadata.go (hunk 1 above)
Comment (DmitriyLewen) [reply metadata/metadata.go:1]: it was miss click, thanks, fixed.

File: pkg/crawler/crawler.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/crawler/crawler.go]: Done

File: pkg/crawler/crawler.go (hunk 4 above)
Comment (knqyf263) [reply pkg/crawler/crawler.go:75]: URL parsing is already ran in goroutine. Why do we need another goroutine?

Comment (DmitriyLewen) [reply pkg/crawler/crawler.go:75]: GORM with SQLite is slow. That is why we need to use separate goroutine.

File: .github/workflows/cron.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Login to GitHub Packages Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ env.GH_USER }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install oras
+        run: |
+          # oras was rollbacked to v0.12.0, because now v0.13.0 (the latest version) contains bugs: https://github.com/oras-project/oras/issues/447
```
Comment (knqyf263) [inline .github/workflows/cron.yml:45]: Seems like this issue is closed. Can you try the newer version?

Comment (DmitriyLewen) [reply .github/workflows/cron.yml:45]: created #4 for updating oras

**Pull request merged**